### PR TITLE
Add option to set headers and sort options in cat requests

### PR DESCRIPTION
**Pull request by MarinaRazumovsky**
### Description
Add option to set headers and sort options in cat requests:

### Issues Resolved
[#388] 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by MarinaRazumovsky** [comment]
Hi @reta. I recreate a new PR  because I made mistake with force merge from main branch and removed all my changes, the previous PR was closed automatically. So I recovered it from my local branch.

File: java-client/src/test/java/org/opensearch/client/opensearch/integTest/OpenSearchJavaClientTestCase.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ protected String getTestRestCluster() {
     protected void wipeAllOSIndices() throws IOException {
         final IndicesResponse response = adminJavaClient()
             .cat()
-            .indices(r -> r.expandWildcards(ExpandWildcard.All));
+            .indices(r -> r.columnNames("index,creation.date").expandWildcards(ExpandWildcard.All));
```
Comment (reta) [inline java-client/src/test/java/org/opensearch/client/opensearch/integTest/OpenSearchJavaClientTestCase.java:157]: I think we changed `columnNames` to `headers`, right?

Comment (MarinaRazumovsky) [reply java-client/src/test/java/org/opensearch/client/opensearch/integTest/OpenSearchJavaClientTestCase.java]: right, fixed it

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,8 +2,6 @@
 Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ## [Unreleased]
-### Dependencies
```
Comment (reta) [inline CHANGELOG.md:5]: Removed by accident?

File: CHANGELOG.md [hunk 3]
```diff
@@ -53,7 +52,6 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Do not double-wrap OpenSearchException on error ([#323](https://github.com/opensearch-project/opensearch-java/pull/323))
 - Fix AwsSdk2TransportOptions.responseCompression ([#322](https://github.com/opensearch-project/opensearch-java/pull/322))
 - Bulk UpdateOperation misses upsert options ([#353](https://github.com/opensearch-project/opensearch-java/pull/353))
-- Fix missing key property in the RangeBucket ([#370](https://github.com/opensearch-project/opensearch-java/pull/370))
```
Comment (reta) [inline CHANGELOG.md:56]: Same?

Comment (MarinaRazumovsky) [reply CHANGELOG.md]: fixed

**Comment by reta** [comment]
@MarinaRazumovsky DCO check is failing :(

**Comment by MarinaRazumovsky** [comment]
@reta what can I do with it? It is because of other commits, that I merged from main

**Comment by reta** [comment]
> @reta what can I do with it? It is because of other commits, that I merged from main

It seems like this is the last commit only:

```
git commit --amend --signoff
```

And force push after, that should help it

**Comment by MarinaRazumovsky** [comment]
@reta as I see in problem description, the issue is in other commits(not mine):
Commit sha: [1fa8ecb](https://github.com/opensearch-project/opensearch-java/pull/412/commits/1fa8ecb49e8fbd44b4e8c41f2751881bea5e3050), Author: Ilya Lukyanovich, Committer: MarinaRazumovsky; Expected "Ilya Lukyanovich [ilya.lukyanovich@dataart.com](mailto:ilya.lukyanovich@dataart.com)", but got "Ilya Lukyanovich [l000kian@gmail.com](mailto:l000kian@gmail.com)".
Commit sha: [7e9da4e](https://github.com/opensearch-project/opensearch-java/pull/412/commits/7e9da4e5f9d161a5643a5338349dd1a0ea82edda), Author: Lucas Rogerio Caetano Ferreira, Committer: MarinaRazumovsky; Expected "Lucas Rogerio Caetano Ferreira [luneo7@users.noreply.github.com](mailto:luneo7@users.noreply.github.com)", but got "luneo7 [luneo7@gmail.com](mailto:luneo7@gmail.com)".

**Comment by reta** [comment]
> @reta as I see in problem description, the issue is in other commits(not mine):

I see, thanks @MarinaRazumovsky, could you try to rebase please?

**Review by reta — APPROVED** [review]

**Review by dblock — DISMISSED** [review]

**Comment by dblock** [comment]
Let's document this in USER_GUIDE?

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Pull request merged**